controller/controller_test.go

```diff
@@ -625,3 +630,67 @@ func checkStats(t *testing.T, r *FakeStatsReporter, reportCount, lastQueueDepth,
 		t.Errorf("Reconcile success = %v, wanted %v", got, want)
 	}
 }
+
+type FixedInformer struct {
+	sunk bool
+}
+
+var _ Informer = (*FixedInformer)(nil)
+
+func (fi *FixedInformer) Run(<-chan struct{}) {}
+func (fi *FixedInformer) HasSynced() bool {
+	return fi.sunk
+}
+
+func TestStartInformersSuccess(t *testing.T) {
+	errCh := make(chan error)
+	defer close(errCh)
+
+	fi := &FixedInformer{sunk: true}
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	go func() {
+		errCh <- StartInformers(stopCh, fi)
+	}()
+
+	select {
+	case err := <-errCh:
+		if err != nil {
+			t.Errorf("Unexpected error: %v", err)
+		}
+	case <-time.After(1 * time.Second):
+		t.Error("Timed out waiting for informers to sync.")
+	}
+}
+
+func TestStartInformersFailure(t *testing.T) {
+	errCh := make(chan error)
+	defer close(errCh)
+
+	fi := &FixedInformer{sunk: false}
+
+	stopCh := make(chan struct{})
+	go func() {
+		errCh <- StartInformers(stopCh, fi)
+	}()
+
+	select {
+	case err := <-errCh:
+		t.Errorf("Unexpected send on errCh: %v", err)
+	case <-time.After(1 * time.Second):
+		// Wait a brief period to ensure nothing is sent.
+	}
+
```

Comment:
Do we want a test, were sunk goes from false -> true? ⬛️

Reply:
Done